src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextEncoder.cs

```diff
@@ -520,19 +520,18 @@ internal void WriteEntityRef(string name)
         // for fragment of a string such as Write( string, offset, count).
         // The string fragment will be written out by copying into a small helper buffer and then
         // calling textWriter to write out the buffer.
-        private void WriteStringFragment(string str, int offset, int count, char[] helperBuffer)
+        private void WriteStringFragment(ReadOnlySpan<char> ros, int offset, int count)
```

Comment:
If you're changing the method to accept a span, why would it need to still accept an offset and a count?